crates/project_model/src/sysroot.rs

```diff
@@ -15,6 +15,7 @@ use crate::utf8_stdout;
 #[derive(Default, Debug, Clone, Eq, PartialEq)]
 pub struct Sysroot {
     crates: Arena<SysrootCrateData>,
+    pub rustc_src_dir: Option<AbsPathBuf>,
```

Comment:
I don't think we need this field here. We can run discovery logic in the `if path_to_rustc_src == "discover" {` branch.